diff --git a/cmake/cpu.cmake b/cmake/cpu.cmake
index ce7a69f..44fdb93 100644
--- a/cmake/cpu.cmake
+++ b/cmake/cpu.cmake
@@ -55,6 +55,11 @@ if(${ANDROID})
   endif()
 endif()
 
+# force set C_FLAGS for arm-linux
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
+ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpu=neon ")
+endif()
+
 list(LENGTH WEBP_SIMD_FLAGS WEBP_SIMD_FLAGS_LENGTH)
 math(EXPR WEBP_SIMD_FLAGS_RANGE "${WEBP_SIMD_FLAGS_LENGTH} - 1")
 unset(HIGHEST_SSE_FLAG)
