diff --git a/tensorflow/compiler/tf2tensorrt/segment/segment.cc b/tensorflow/compiler/tf2tensorrt/segment/segment.cc
index b3886f3e..fcb1fa95 100644
--- a/tensorflow/compiler/tf2tensorrt/segment/segment.cc
+++ b/tensorflow/compiler/tf2tensorrt/segment/segment.cc
@@ -257,8 +257,9 @@ void StableDFS(const SimpleGraph& g, bool reverse,
     stack[i] = Work{start[i], false};
   }
 
-  auto get_nodes = reverse ? [](const SimpleNode* n) -> std::vector<SimpleNode*> { return n->in_nodes(); }
-                           : [](const SimpleNode* n) -> std::vector<SimpleNode*> { return n->out_nodes(); };
+  auto get_nodes = [reverse](const SimpleNode* n) { 
+    return reverse ? n->in_nodes() : n->out_nodes(); 
+  };
   std::vector<bool> visited(g.num_node_ids(), false);
   while (!stack.empty()) {
     Work w = stack.back();
