diff --git a/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc b/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc
index 8083a554..fb6bdfe6 100644
--- a/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc
+++ b/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc
@@ -353,7 +353,7 @@ Status GetTrtBroadcastShape(const TRT_TensorOrWeights& operand_l,
   }
 
   const int max_nb_dims = nvinfer1::Dims::MAX_DIMS + 1;
-  auto compute_output_dims = [use_implicit_batch](
+  auto compute_output_dims = [use_implicit_batch, max_nb_dims](
                                  const TRT_TensorOrWeights& input,
                                  int broadcast_num_dims, int* output_dims_array,
                                  nvinfer1::Dims* output_dims) {
